core/src/test/scala/org/apache/spark/sql/delta/DeltaTestUtils.scala

```diff
@@ -116,7 +135,14 @@ trait DeltaTestUtilsBase {
   }
 }
 
-object DeltaTestUtils extends DeltaTestUtilsBase
+object DeltaTestUtils extends DeltaTestUtilsBase {
+  case class Plans(
```

Comment:
this seems generally useful. why not add it to the base?

Reply:
It could cause type mismatches if test suites extend the base trait and use this class. Functions inside `DeltaTestUtilsBase` will expect `Plans` to be defined `DeltaTestUtilsBase`, and functions defined in test suites will expect `Plans` to be defined in their suite. Putting this inside the singleton will make sure there's a single definition all users reference.